### DISCUSSION TITLE
"play once" option to prevent scheduled playlists from restarting if they reache  the end too soon

### DISCUSSION THREAD

**Discussion by silvaalex2021** [AzuraCast/AzuraCast#4186] [Ideas]
Helo!

I have been facing a situation since I started using azuracast. It didn't really matter at first because I was just playing and testing, but now it's starting to bother me ...

I have a handful of scheduled playlists, which are pre-recorded shows. Usually I schedule the playlist for the duration of the show, for example if the show lasts 120 mins I will schedule the playlist with 120 mins between the "start" and "end" time.

sometimes the show will end a little earlier, and this will make the scheduled playlist start again, restarting the show instead of switching to the general rotation as it should be.

For example, if I have a playlist scheduled to play between 8 pm and 10 pm, and the show ends early, for example at 9:50 pm or 9:45 pm instead of exactly at 10:00 pm, this will cause the show to be rescheduled and, if it is a single file, it will play again until the end.

I think there should definitely be some kind of configuration in the playlist to avoid this, like an option that tells Azuracast AutoDJ to run through that playlist scheduled only once until the end.

On the "advanced" tab of the playlist settings dialog there is an option that does just that, but it is already known that it does not work well in conjunction with the azuracast autoDJ.

is there anything i can do about it?

Thanks in advance!

**Comment by darrylcarter** [comment]
If it's one file just set it to play for one minute. That's my work around anyway

**Reply by silvaalex2021** [reply]
it is not a single file. Usually, there are 4 or 6 files, with 12, 13 minutes each.

**Comment by Vaalyn** [comment]
@silvaalex2021 I have added a new option to each schedule item to make it only loop through a playlist once at the scheduled time called `Loop Once`:

![image](https://user-images.githubusercontent.com/13745863/125344731-f19a1900-e357-11eb-8e4c-de6981adcfdb.png)

When the AutoDJ checks if the playlist should be played at the beginning of the scheduled time it will make sure that the playback queue is full / reset right before it plays so that it plays all contents. When the playlists internal playback queue is empty (due to all tracks having been played) or was reset in the scheduled time (due to the playlists internal queue being empty) and has already been played between start and end it will not loop it again in that scheduled time block.

Would you mind helping us test this new feature?
